0.40

balloob opened the pull request:
It's time for version 0.40 already! This version the main focus was on performance and bug fixes. Big thanks to [@pvizeli] for driving this effort. Startup is now super fast. We have also continued with adding warnings for components and platforms that are slowing down Home Assistant.

Some people have interpreted our recent added warnings as if 0.39 introduced the issues that caused the warning. This is not the case, the platforms and components have been misbehaving for a while but now we are adding warnings so we are able to track down the culprits.

Before we continue talking about all the improvements in this release I want to take a moment to thank a few people from the Home Assistant community that are fundamental to the success of Home Assistant. Big thanks to [@dale3h], [@CCOSTAN], [@skalavala], [@rrubin0], [@brahmafear], [@bassclarinetl2] and [@torn8o]! These are all people that hang out in our [main chat channel] and help new users get started and help existing users when running into trouble. Home Assistant would not be there without all the effort you put in! 🙇 

## Z-Wave is now threadsafe

[@andrey-git] has spend a lot of time to make Z-Wave perform better. It should no longer cause Home Assistant to run slower or raise warnings about the timer getting out of sync.

## Update on the MQTT Out of Memory errors in 0.39

MQTT started causing "Out of Memory" errors for some people on a Raspberry Pi. We have been able to track this down to Raspberry Pis that are using an older firmware. If you are experiencing this issue, please [upgrade your firmware using `rpi-update`](https://github.com/Hexxeh/rpi-update#installing).

## Android IP Camera

With the new support for [IP Webcam](https://play.google.com/store/apps/details?id=com.pas.webcam) added by [@robbiet480] and [@pvizeli] you are now able to re-purpose any Android phone to become a multifunctional IP webcam.

 - Add the Android device camera as a camera in Home Assistant
 - Binary sensor when motion is detected
 - Sensors to expose the device sensors, including pressure, sound, battery, light
 - Control device features like the GPS, night vision and camera flash

![Default layout](http://d.pr/i/dcm3+)

## Other Highlights

 - Support added for Austrian weather using Zamg weather data ([@Zac-HD])
 - AppleTV has been converted to push data to Home Assistant instead of us having to poll ([@postlund])
 - Ring.com video doorbell integration added ([@tchellomello])
 - Blink Home Security Camera support added ([@fronzbot])
 - MQTT discovery now works for lights. ([@robbiet480])
 - The MQTT light platform now has `effect`, `white_value` and `xy_color` support. ([@robbiet480])
 - The MQTT JSON light platform now has `color_temp`, `effect`, `white_value` and `xy_color` support. ([@robbiet480])
  - The MQTT Template light platform now has `color_temp`, `white_value` and `xy_color` support. ([@robbiet480])

## Breaking changes

- Vera entity ids have changed. This is a one time change to migrate to a model that will prevent future conflicts.
- The Twilio notify platforms now have to be configured via the twilio component. 

```yaml
twilio:
  account_sid: "abc"
  auth_token: "xyz"
```

- If you are using async custom components, the passed in `async_add_devices` method is now a callback instead of a coroutine.

## All changes

- Cleanup run_callback_threadsafe ([@pvizeli])
- Use H2 headers to split up the different sections ([@colinodell])
- Refactory of envisalink ([@pvizeli])
- Create zwave devices on OZW thread and only add them during discovery ([@andrey-git])
- Bugfix restore startup state ([@kellerza])
- Random test fixes ([@balloob])
- Remove automatically reloading group config ([@balloob])
- Default config to setup group editor ([@balloob])
- minor broadlink fix ([@danielhiversen])
- Update Yeelight Sunflower light platform to 0.0.6 ([@lindsaymarkwawrd])
- Some zwave cleanup ([@andrey-git])
- sensor.speedtest: provide a default icon ([@molobrakos])
- Test the temperature returned by RM2 ([@aronsky])
- Zamg weather ([@Zac-HD])
- Fix reporting on bad login ([@balloob])
- Move mqtt from eventbus to dispatcher / add unsub for dispatcher ([@pvizeli])
- Update flake8 and pylint to latest ([@andrey-git])
- Fix link ([@fabaff])
- Make glob preserve order ([@andrey-git])
- Update regex ([@fabaff])
- Fix recorder async ([@balloob])
- Fix livebox-play interactions for Python < 3.6 ([@pschmitt])
- Ensure we properly close HASS instances. ([@balloob])
- Add service to change log levels ([@postlund])
- Move ffmpeg to dispatcher from hass.data entity store. ([@pvizeli])
- Feature/reorg recorder ([@balloob])
- Bugfix mqtt socket error ([@pvizeli])
- Notify ciscospark ([@shenning00])
- Config fix ([@balloob])
- Bugfix mqtt paho client to speend time ([@pvizeli])
- Properly report features for each hue bulb type ([@jawilson])
- Local file camera now supports yet inexisting files. ([@jjmontestl])
- light.transition now supports float instead of int in order to be able to perform faster transitions ([@BillyNate])
- Fix for OSRAM lights connected to hue bridge ([@groth-its])
- Add support for MAX!Cube thermostats and window shutter sensors ([@BastianPoe])
- Analog modem callerid support ([@vroomfonde1])
- [sensor.dnsip] New Sensor: DNS IP ([@danielperna84])
- Update library version for Yeelight Sunflower lights platform (fix for packaging problem with 0.0.7) ([@lindsaymarkwawrd])
- Prevent duplicate names on Vera devices by appending the device id ([@arjenfvellinga])
- Add temperature support for MH-Z19 CO2 sensor. ([@andrey-git])
- improve history_stats accuracy ([@bokub])
- Updated pyitachip2ir ([@alanfischer])
- Influx fix ([@open-homeautomation])
- Fix toggle and media_play_pause post async ([@armills])
- Migrate calendar setup to async. ([@pvizeli])
- Frontier silicon ([@zhelev])
- Bootstrap / Component setup async ([@pvizeli])
- Convert kpH and mpH to kph and mph ([@ericgingras])
- Rollback netdisco to 0.8.2 to resolve #6165 ([@jumpkick])
- Log errors when loading yaml ([@kellerza])
- Bootstrap tweaks tests ([@balloob])
- Telegram webhooks new text event ([@scipioni])
- Cleanup component track_point_in_utc_time usage ([@pvizeli])
- Discovery fix ([@balloob])
- Test against 3.6-dev ([@balloob])
- Bugfix ZigBee / Move from eventbus to dispatcher ([@pvizeli])
- Bump netdisco to 0.9.1 ([@balloob])
- sensor.dovado: compute state in update ([@molobrakos])
- Fix mysensors callback race ([@MartinHjelmare])
- Upgrade TwitterAPI to 2.4.5 ([@fabaff])
- Upgrade py-cpuinfo to 0.2.6 ([@fabaff])
- Template sensor change flow / add restore ([@pvizeli])
- Zwave optimize value_added ([@andrey-git])
- Update Vagrant provision.sh ([@shaftoe])
- Update Adafruit_Python_DHT to support new raspberry kernel ([@masarliev])
- Add fallback for name if userdevicename isn't set using old serialnumber logic ([@reedriley])
- Improve Honeywell US climate component ([@titilambert])
- Template binary_sensor change flow / add restore ([@pvizeli])
- Additional support for ecobee hold mode ([@Duoxilian])
- Update Formulas in Convert XY to RGB ([@dramamoose])
- Use dynamic ports for test instances ([@armills])
- Added support for multiple codes executed in a row ([@martinfrancois])
- Use push updates in Apple TV ([@postlund])
- Fix command sudo not found error in dev Dockerfile ([@jawilson])
- Fix calendar authentication text, and handle calendar events without summaries. ([@alanfischer])
- Move dispatcher out of init. ([@pvizeli])
- Zwave: Add remove/replace failed node services. ([@andrey-git])
- Template switch change flow / add restore ([@pvizeli])
- Bump limitlessled dependency to 1.0.5. ([@janLo])
- snmp: upgrade pysnmp to 4.3.4 ([@milaq])
- Bugfix new async_add_devices function ([@pvizeli])
- Restore for input_slider ([@pvizeli])
- Added IPv4 data collector ([@open-homeautomation])
- Return None instead of raising ValueException from as_timestamp template function. ([@jjmontestl])
- [recorder] Catch more startup errors #6179 ([@kellerza])
- twilio component ([@happyleavesaoc])
- Add Z-Wave battery level as a sensor. ([@andrey-git])
- OwnTrack Async ([@pvizeli])
- Fix possibility that have multible topic subscribe mqtt ([@pvizeli])
- Migrate mqtt tracker and arwn sensor to async / cleanup owntrack ([@pvizeli])
- Z-Wave prevent I/O event loop ([@balloob])
- Update pwaqi to 3.0 to use public API ([@valentinalexeev])
- Update Hikvision Binary Sensors to latest library, remove pyDispatcher ([@mezz64])
- Don't initialize components which have already been discovered ([@colinodell])
- Comed Hourly Pricing sensor ([@joe248])
- Add multi contracts support for Hydroquebec ([@titilambert])
- Add Zwave refresh services ([@andrey-git])
- Add keep-alive feature to the generic thermostat ([@aronsky])
- Fix wake_on_lan for german version of Windows 10 (#6397) ([@siebert])
- flux led lib ([@danielhiversen])
- Cleanup async handling ([@pvizeli])
- Restore for automation entities ([@kellerza])
- Fix tests no internet ([@balloob])
- Prevent more I/O in apns ([@balloob])
- Restore flow on device_tracker platform ([@pvizeli])
- switch.tplink: catch exceptions coming from pyHS100 to avoid flooding the logs when the plug is not available ([@rytilahti])
- Added sensors to support Ring.com devices ([@tchellomello])
- Split bootstrap into bs + setup ([@balloob])
- Tweak recorder/restore_state ([@balloob])
- Fix unnecessary warning for ip bans.yaml ([@balloob])
- Better restore_state warnings ([@balloob])
- Set new color before turning LIFX bulbs on ([@amelchio])
- Don't log username and password in camera url ([@ishults])
- Ignore deleted mails in IMAP unread count (#6394) ([@amelchio])
- Delay zwave updates for 100ms to group them. ([@andrey-git])
- Rename _scheduled_update to _update_scheduled ([@andrey-git])
- Revert "Use dynamic port allocation for tests" ([@armills])
- Tado device_tracker exception when mobile device has geofencing enabled but location is currently unknown. ([@jmvermeulen])
- Add a Z-wave workaround to do full refresh on update ([@andrey-git])
- Use bundled certificates if port matches mqtts ([@dennisdegreef])
- Bugfix samsungtv discovery ([@pvizeli])
- Added unittest for Ring sensor ([@tchellomello])
- Shorten recorder connection init ([@balloob])
- KWB Easyfire support ([@bimbar])
- Bumped version number for supporting lib ([@bazwilliams])
- Send a logo with webostv notifications ([@andersonshatch])
- Upgrade netdisco to 0.9.2 ([@balloob])
- Allow testing against uvloop ([@balloob])
- fix issue ([@appzer])
- Remove connection status state. ([@aequitas])
- Support for Blink Camera System ([@fronzbot])
- Add warning for slow platforms/components ([@balloob])
- Fix wake_on_lan ping for Linux. ([@siebert])
- Add dispatcher camera for internal image. ([@pvizeli])
- Add support for remove services / Reload script support ([@pvizeli])
- Expand MQTT lights ([@robbiet480])
- Allow configurable conditions for Pi-Hole sensor ([@colinodell])
- Improved iCloud 2FA support. ([@reedriley])
- Update pymyq requirement ([@arraylabs])
- Not always asume manufacturername is present ([@balloob])
- Add first pass at Z-Wave light tests ([@balloob])
- Bugfix mqtt socket memory error ([@pvizeli])
- Increase upper limit on light transitions ([@amelchio])
- Bugfix android camera autodiscovery settings ([@pvizeli])
- Update to Pyunifi2.0 ([@finish06])
- Insteon lib ([@wardcraigj])
- Bugfix rpi_rf cleanup ([@pvizeli])
- Android webcam better error handling / pump library 0.4 ([@pvizeli])

[@BastianPoe]: https://github.com/BastianPoe
[@BillyNate]: https://github.com/BillyNate
[@Duoxilian]: https://github.com/Duoxilian
[@MartinHjelmare]: https://github.com/MartinHjelmare
[@Zac-HD]: https://github.com/Zac-HD
[@aequitas]: https://github.com/aequitas
[@alanfischer]: https://github.com/alanfischer
[@amelchio]: https://github.com/amelchio
[@andersonshatch]: https://github.com/andersonshatch
[@andrey-git]: https://github.com/andrey-git
[@appzer]: https://github.com/appzer
[@arjenfvellinga]: https://github.com/arjenfvellinga
[@armills]: https://github.com/armills
[@aronsky]: https://github.com/aronsky
[@arraylabs]: https://github.com/arraylabs
[@balloob]: https://github.com/balloob
[@bazwilliams]: https://github.com/bazwilliams
[@bimbar]: https://github.com/bimbar
[@bokub]: https://github.com/bokub
[@colinodell]: https://github.com/colinodell
[@danielhiversen]: https://github.com/danielhiversen
[@danielperna84]: https://github.com/danielperna84
[@dennisdegreef]: https://github.com/dennisdegreef
[@dramamoose]: https://github.com/dramamoose
[@ericgingras]: https://github.com/ericgingras
[@fabaff]: https://github.com/fabaff
[@finish06]: https://github.com/finish06
[@fronzbot]: https://github.com/fronzbot
[@groth-its]: https://github.com/groth-its
[@happyleavesaoc]: https://github.com/happyleavesaoc
[@ishults]: https://github.com/ishults
[@janLo]: https://github.com/janLo
[@jawilson]: https://github.com/jawilson
[@jjmontestl]: https://github.com/jjmontestl
[@jmvermeulen]: https://github.com/jmvermeulen
[@joe248]: https://github.com/joe248
[@jumpkick]: https://github.com/jumpkick
[@kellerza]: https://github.com/kellerza
[@lindsaymarkwawrd]: https://github.com/lindsaymarkwawrd
[@martinfrancois]: https://github.com/martinfrancois
[@masarliev]: https://github.com/masarliev
[@mezz64]: https://github.com/mezz64
[@milaq]: https://github.com/milaq
[@molobrakos]: https://github.com/molobrakos
[@open-homeautomation]: https://github.com/open-homeautomation
[@postlund]: https://github.com/postlund
[@pschmitt]: https://github.com/pschmitt
[@pvizeli]: https://github.com/pvizeli
[@reedriley]: https://github.com/reedriley
[@robbiet480]: https://github.com/robbiet480
[@rytilahti]: https://github.com/rytilahti
[@scipioni]: https://github.com/scipioni
[@shaftoe]: https://github.com/shaftoe
[@shenning00]: https://github.com/shenning00
[@siebert]: https://github.com/siebert
[@tchellomello]: https://github.com/tchellomello
[@titilambert]: https://github.com/titilambert
[@valentinalexeev]: https://github.com/valentinalexeev
[@vroomfonde1]: https://github.com/vroomfonde1
[@wardcraigj]: https://github.com/wardcraigj
[@zhelev]: https://github.com/zhelev
[main chat channel]: https://gitter.im/home-assistant/home-assistant
[@dale3h]: https://github.com/dale3h
[@CCOSTAN]: https://github.com/CCOSTAN
[@skalavala]: https://github.com/skalavala
[@rrubin0]: https://github.com/rrubin0
[@brahmafear]: https://github.com/brahmafear
[@bassclarinetl2]: https://github.com/bassclarinetl2
[@torn8o]: https://github.com/torn8o

deftdawg committed:
Improvements for WeMo Insight switches

* Changes current power units to watts
* Adds power on times and additional totals

deftdawg committed:
Update wemo.py

* Reordered datetime import
* Spaces by 4

deftdawg committed:
Update wemo.py

* continuation line under-indented for visual indent

deftdawg committed:
Update wemo.py

trailing whitespace... (argh... the bot should just trim it)

deftdawg committed:
Back to you travis....

deftdawg committed:
Update wemo.py

deftdawg committed:
Update wemo.py

deftdawg committed:
Update wemo.py

kellerza committed:
Allow 4.5min startup time for recorder

deftdawg committed:
Update device_state_attributes only

This gets rid of the other stuff and just updates device_state_attributes, leaving the default properties alone.

pvizeli committed:
Cleanup run_callback_threadsafe (#6187)

* Cleanup run_callback_threadsafe

* fix spell

* Revert image_processing, they need to wait for update

colinodell committed:
Use H2 headers to split up the different sections (#6183)

Using headers makes it easier to visually differentiate between the different sections

pvizeli committed:
Refactory of envisalink (#6160)

* Refactory of envisalink

* remove event buss

* init dispatcher from hass.

* Move platform to new dispatcher

* fix lint

* add unittest & threadded functions

* fix copy & past error

andrey-git committed:
Create zwave devices on OZW thread and only add them during discovery (#6096)

* Create zwave devices on OZW thread and only add them during discovery.

* Read and write devices dict from loop thread.

* More async

* replace callback with coroutine

* import common function instead of callin git

deftdawg committed:
Incorporate comment suggestions

- Separate attribs from coffeemaker condition
- Set power units for threshold to mW to be consistent with others
- Adjust on-time labels to be more clear

kellerza committed:
Bugfix restore startup state (#6189)

balloob committed:
Random test fixes (#6195)

* Store persistent errors in hass (speeds up tests)

* Fix sleepiq test dependency on test order

* Fix sleepiq validation

balloob committed:
Extend test for group config

balloob committed:
Remove automatically reloading group config (#6197)

balloob committed:
Default config to setup group editor (#6198)

Danielhiversen committed:
minor broadlink fix (#6202)

lindsaymarkward committed:
Update Yeelight Sunflower light platform to 0.0.6 (#6208)

Add an optional extended description…

andrey-git committed:
Some zwave cleanup (#6203)

molobrakos committed:
sensor.speedtest: provide a default icon (#6207)

aronsky committed:
Test the temperature returned by RM2 (#6205)

* Test the temperature returned by RM2
* Validate fields via voluptuous
* Fixed range for humidity

Zac-HD committed:
Zamg weather (#5894)

* Fast & efficient updates for ZAMG weather data

ZAMG updates on the hour, so instead of checking every half-hour we can
check each minute - only after the observations are taken until
receiving them.

* sensor.zamg: test instead of whitelist for station_id

* Autodetect closest ZAMG station if not given

* ZAMG weather component, based on the sensor

* Review improvements

* Update to new ZAMG schema, add logging

Turns out it wasn't a typo, but rather an upstream schema change.  Added
better error handling to ease diagnosis in case it happens again.

* No hardcoded name

pavoni committed:
Bump pyloopenergy - catch socketIO exceptions.

pavoni committed:
Merge pull request #6213 from home-assistant/bump_pyloopenergy

Bump pyloopenergy - catch socketIO exceptions.

balloob committed:
Fix reporting on bad login (#6201)

pvizeli committed:
Move mqtt from eventbus to dispatcher / add unsub for dispatcher  (#6206)

* Move mqtt from eventbus to dispatcher / add unsub for dispatcher

* Fix lint

* Fix test

* Fix lint v2

* fix dispatcher_send

andrey-git committed:
Update flake8 and pylint to latest (#6217)

molobrakos committed:
sensor.dovado: Upgraded library version

molobrakos committed:
sensor.eliqonline: Change icon

fabaff committed:
Fix link (#6219)

kellerza committed:
No wait for start and more async

andrey-git committed:
Make glob preserve order (#6224)

fabaff committed:
Update regex (#6216)

balloob committed:
Merge pull request #6167 from kellerza/recorder_timer

Recorder async & remove start_recording event

balloob committed:
Update frontend

balloob committed:
Fix recorder async (#6228)

pschmitt committed:
Fix livebox-play interactions for Python < 3.6 (#6243)

balloob committed:
Ensure we properly close HASS instances. (#6234)

postlund committed:
Add service to change log levels (#6221)

* Add service to change log levels

* Fix review comments

pvizeli committed:
Move ffmpeg to dispatcher from hass.data entity store. (#6211)

* Move ffmpeg to dispatcher from hass.data entity store.

* fix lint

* address paulus comments

* add more unittest for better coverage

balloob committed:
Feature/reorg recorder (#6237)

* Re-organize recorder

* Fix history

* Fix history stats

* Fix restore state

* Lint

* Fix session reconfigure

* Move imports around

* Do not start recording till HASS started

* Lint

* Fix logbook

* Fix race condition recorder init

* Better reporting on errors

pvizeli committed:
Bugfix mqtt socket error (#6256)

shenning00 committed:
Notify ciscospark (#6130)

* Adding ciscospark notifier

* Adding ciscospark notifier

* CI cleanup.

* houndci-bot changes

* ok --- a bunch of code verify changes

balloob committed:
Config fix (#6261)

pvizeli committed:
Bugfix mqtt paho client to speend time (#6266)

jawilson committed:
Properly report features for each hue bulb type (#6271)

jjmontesl committed:
Local file camera now supports yet inexisting files. (#6157)

BillyNate committed:
light.transition now supports float instead of int in order to be able to perform faster transitions (#6163)

groth-its committed:
Fix for OSRAM lights connected to hue bridge (#6122)

* Fix for OSRAM lights connected to hue bridge

Do not send command "effect = none" to OSRAM lights
Osram lights connected to a hue bridge do not seem to handle "effect =
none" very well. Most of the times they jump to the selected color and
then change to red within a second.

Osram lights connected to a hue bridge do not handle xy values outside
of their gamut. Since they just stay at their old color value, handling
the UI is very unpredictable. Sending HSV values to the lights fixes this.

* Add tests for new util methods

BastianPoe committed:
Add support for MAX!Cube thermostats and window shutter sensors (#6105)

vroomfonde1 committed:
Analog modem callerid support (#5840)

* analog-modem-callerid

* analog-modem-callerid

* analog-mod

* Updates from latest review

* Updates from latest review

pavoni committed:
Fix vera thermostat mode set bug

danielperna84 committed:
[sensor.dnsip] New Sensor: DNS IP (#6214)

* Added DNS IP sensor

* Removed unused import

* Added coverage

* fixed flake

* Applied suggested changes

* Removed debug code

* Switched to aiodns

* Raised scan interval

* Updating state with entity creation

* Lint

* Updated requirements_all

pavoni committed:
Merge pull request #6278 from home-assistant/fix_vera_thormostat_bug

Fix vera thermostat bug

lindsaymarkward committed:
Update library version for Yeelight Sunflower lights platform (fix for packaging problem with 0.0.7) (#6233)

arjenfvellinga committed:
Prevent duplicate names on Vera devices by appending the device id (#6100)

* Prevent duplicate names by prepending device id to it.

* Always append device id, not conditionally.

* Moved naming of devices

* flake8

andrey-git committed:
Add temperature support for MH-Z19 CO2 sensor. (#6169)

* Add temperature support for MH-Z19 CO2 sensor.

* Remove debug printout

* More tests

* Minor fixes

bokub committed:
improve history_stats accuracy (#6294)

alanfischer committed:
Updated pyitachip2ir (#6296)

* Updated pyitachip2ir

* Updated requirements_all.txt

Commit:
Influx fix (#6289)

* Fix: replace influxdb query by another query that is more lightweight and won't timeout

* Fix: replace influxdb query by another query that is more lightweight and won't timeout

emlove committed:
Fix toggle and media_play_pause post async (#6291)

pvizeli committed:
Migrate calendar setup to async. (#6305)

zhelev committed:
Frontier silicon (#6131)

* added frontier_silicon constant

* added the frontier_silicon component

* cleaning up according to travis

* trying to satisfy pylint

* trying to satisfy pylint

* fsapi version 0.0.6

* with fsapi version 0.0.7

* added fsapi dependency

* yielding the FSAPI

* Removing white space from docstring

* Removing white space from an empty line

* Switching to sync

* clean up white spaces and rename device to FSAPIDevice

* added frontier_silicon constant

* added the frontier_silicon component

* cleaning up according to travis

* trying to satisfy pylint

* trying to satisfy pylint

* fsapi version 0.0.6

* with fsapi version 0.0.7

* added fsapi dependency

* yielding the FSAPI

* Removing white space from docstring

* Removing white space from an empty line

* Switching to sync

* clean up white spaces and rename device to FSAPIDevice

* changed info to debug

* added frontier_silicon constant

* added the frontier_silicon component

* cleaning up according to travis

* trying to satisfy pylint

* trying to satisfy pylint

* fsapi version 0.0.6

* with fsapi version 0.0.7

* added fsapi dependency

* yielding the FSAPI

* Removing white space from docstring

* Removing white space from an empty line

* Switching to sync

* clean up white spaces and rename device to FSAPIDevice

* added the frontier_silicon component

* trying to satisfy pylint

* fsapi version 0.0.6

* remove white space

* generated requirements

* added the frontier_silicon component

* cleaning up according to travis

* trying to satisfy pylint

* trying to satisfy pylint

* fsapi version 0.0.6

* with fsapi version 0.0.7

* added fsapi dependency

* yielding the FSAPI

* Removing white space from docstring

* Removing white space from an empty line

* Switching to sync

* clean up white spaces and rename device to FSAPIDevice

* trying to satisfy pylint

* changed info to debug

* added the frontier_silicon component

* fsapi version 0.0.6

* generated requirements

* pylint

* moved import requests to the method where it is being used

* add a basic unit test

* cleaned up source code

* added frontier_silicon constant

* added the frontier_silicon component

* added basic test

* added fsapi to requirements_all.txt

* added coverage omit, though a basic test was included

* added MEDIA_TYPE_MUSIC for artist and album

* removed duplicate cons

* switched fsapi call to a property, removed unecessary comment

* detailed docstring for fs_device

* added a space for the info_name - info_text separator

* reduced proeprty (fsapi) access for volume down/up

emlove committed:
Version bump to 0.40.0.dev0

pvizeli committed:
Bootstrap / Component setup async (#6264)

* Bootstrap / Entiy setup async

* Cleanup add_job stuff / return task/future object

* Address paulus comments / part 1

* fix install pip

* Cleanup bootstrap / move config stuff to config.py

* Make demo async

* Further bootstrap improvement

* Address Martin's comments

* Fix initial tests

* Fix final tests

* Fix bug with prepare loader

* Remove no longer needed things

* Log error when invalid config

* More cleanup

* Cleanups platform events & fix lint

* Use a non blocking add_entities callback for platform

* Fix Autoamtion is setup befor entity is ready

* Better automation fix

* Address paulus comments

* Typo

* fix lint

* rename functions

* fix tests

* fix test

* change exceptions

* fix spell

e850205 committed:
Convert kpH and mpH to kph and mph (#6316)

There is no reason for the H to be capitalized. Changing it to lowercase increases consistency with other components and allows for use of the min/max sensor, which throws an error if the units of measurement are not the same.

deftdawg committed:
Rollback netdisco to 0.8.2 to resolve #6165 (#6314)

* Rollback netdisco to 0.8.2 to resolve #6165

* Rollback netdisco to 0.8.2 to resolve #6165

kellerza committed:
Log errors when loading yaml (#6257)

balloob committed:
Bootstrap tweaks tests (#6326)

* Update strings/fix component not found message.

* Fix tests

* More tweak text

molobrakos committed:
Merge pull request #6328 from molobrakos/eliqonline

sensor.eliqonline: Change to more appropriate icon

molobrakos committed:
Merge pull request #6329 from molobrakos/dovado

sensor.dovado: Upgraded library version

scipioni committed:
Telegram webhooks new text event (#6301)

* new TELEGRAM_TEXT

* telegram command event renamed in 'telegram_command'

* fire telegram_text event anyway

pvizeli committed:
Cleanup component track_point_in_utc_time usage (#6330)

balloob committed:
Discovery fix (#6321)

* Fix incorrect import

* Create own discovery service

* Fix tests

* Fix hdmi_cec bad import

balloob committed:
Test against 3.6-dev (#6324)

pvizeli committed:
Bugfix ZigBee / Move from eventbus to dispatcher (#6333)

* Bugfix ZigBee / Move from eventbus to dispatcher

* fix lint

balloob committed:
Bump netdisco to 0.9.1 (#6338)

molobrakos committed:
sensor.dovado: compute state in update (#6340)

MartinHjelmare committed:
Fix mysensors callback race (#6311)

* Fix possible race at startup in mysensors callback

* Update devices via persistence before starting gateway to avoid
  two threads calling the same callback at the same time.

* Call add_devices max once per callback

fabaff committed:
Upgrade TwitterAPI to 2.4.5 (#6351)

fabaff committed:
Upgrade py-cpuinfo to 0.2.6 (#6335)

pvizeli committed:
Template sensor change flow / add restore (#6336)

andrey-git committed:
Zwave optimize value_added (#6210)

* Make zwave devices listen on less network changes.

* Convert more platforms

* Remove printouts.

* Fix copy-paste

* Change default dependent list to empty list

Commit:
Update Vagrant provision.sh (#6236)

- Bugfix: with f63a79ee we removed `script/home-assistant@.service`
  systemd unit file, which is used by Vagrant box to start/stop hass
- simplify interaction with Vagrant, provision.sh now is the only
  entry point and doesn't need the user to touch/remove files in
  order to change provisioner behavior

masarliev committed:
Update Adafruit_Python_DHT to support new raspberry kernel (#6325)

* Update Adafruit_Python_DHT to support new raspberry kernel

* update Adafruit Python DHT

reedriley committed:
Add fallback for name if userdevicename isn't set using old serialnumber logic (#6265)

Add an optional extended description…

balloob committed:
Merge remote-tracking branch 'origin/master' into dev

titilambert committed:
Improve Honeywell US climate component (#5313)

* Improve Honeywell US climate component

* Fix tests

* Fix tests

* Add cool_away_temp and heat_away_temp for honeywell US

* Fix honeywell tests

* Fix PR comments

pvizeli committed:
Template binary_sensor change flow / add restore (#6343)

* Template binary_sensor change flow / add restore

* fix lint

Duoxilian committed:
Additional support for ecobee hold mode (#6258)

* Integrate suggestion in #5590 by nordlead2005. This change has been
sitting in limbo for over a month, but it is a good idea. I don't
mean to step on nordlead2005's toes, but we need to make progress.

* Use defined constant for TEMPERATURE_HOLD

* Integrate handling of vacation into hold mode. Canceling vacation
hold requires an update to the external pyecobee library. Creation
of vacation is not supported (it would be straightforward in the code,
but a complex user interface would be required, similar to what is
now done in the ecobee thermostat).

* Add capability to retrieve list of defined climates from ecobee.

* The mode() method used to return the system mode in internal
representation. However, the user sees a different notation in
the ecobee thermostat. Seeing some internal name is particularly
weired with user-defined climates, where these are named "smart1",
"smart2", etc., instead of the name the user has defined. Return
the user-defined name instead. This change might break some user
interfaces but is easily remedied (e.g., use "Away" instead of
"away").

* Simplify is_away_mode_on().

* Correction of erroneously indented else statement.

* Change comment as flake8 gets confused.

dramamoose committed:
Update Formulas in Convert XY to RGB (#6322)

* Update to Current RGB D65 Conversion

As per Philips Hue https://developers.meethue.com/documentation/color-conversions-rgb-xy

* Update the source of the XYZ to RGB formulas

* Fix Whitespace

* Update Whitespace

* Update Tests for new Formulas

* Update Tests

* Update XY_Brightness_to_hsv tests

* Update test_color.py

balloob committed:
Update coveragerc

emlove committed:
Use dynamic ports for test instances (#6232)

martinfrancois committed:
Added support for multiple codes executed in a row (#5908)

* Added support for multiple codes executed in a row

now codes can be specified either by simply providing a single code, which will then be sent like usual, or multiple codes can be executed in a row, specified in a comma delimited format in the configuration.yaml. For example: 111111,222222,333333,444444 would mean 111111 would be sent first, followed by 222222 and 333333 and 444444.

* rpi_rf: added line breaks to not exceed 79 characters per line

* include validation for correct formatting of codes

added regex which only allows either a single number (like 1252456245) or a sequence of commas followed by another number.

* added line breaks to not exceed 79 characters per line

* fix for 'continuation line under-indented for visual indent'

* another try at 'continuation line under-indented for visual indent'

* changed from regex to list for easier maintainability

* removed unnecessary splitting of strings

postlund committed:
Use push updates in Apple TV (#6323)

* Use push updates in Apple TV

* Fix review comments

jawilson committed:
Fix command sudo not found error in dev Dockerfile (#6346)

alanfischer committed:
Fix calendar authentication text, and handle calendar events without summaries. (#6337)

* Fixed google authorization text

* Let calendar handle events without a summary

pvizeli committed:
Move dispatcher out of init. (#6355)

andrey-git committed:
Zwave: Add remove/replace failed node services. (#6248)

* Zwave: Add remove/replace failed node services.

* Fix text

pvizeli committed:
Template switch change flow / add restore (#6356)

* Template switch change flow / add restore

* fix tests

* fix binary_sensor template

janLo committed:
Bump limitlessled dependency to 1.0.5. (#6334)

This fixes issue #6295.

GreenTurtwig committed:
Updated to catch timeout error

milaq committed:
snmp: upgrade pysnmp to 4.3.4 (#6359)

* snmp: upgrade pysnmp to 4.3.4

fixes https://github.com/home-assistant/home-assistant/issues/6238

* snmp: v4.3.4: add missing definition changes

pvizeli committed:
Bugfix new async_add_devices function (#6362)

pvizeli committed:
Restore for input_slider (#6360)

Commit:
Added IPv4 data collector (#6304)

* Added IPv4 data collector

* Formatting

* Bugfix: data is in kBit/s not kByte/s

jjmontesl committed:
Return None instead of raising ValueException from as_timestamp template function. (#6155)

kellerza committed:
[recorder] Catch more startup errors #6179 (#6192)

* [recorder] Catch more startup errors #6179

* Rebase on new recorder

happyleavesaoc committed:
twilio component (#6348)

* twilio component

* add http dependency to twilio

* fire->async_fire

andrey-git committed:
Add Z-Wave battery level as a sensor. (#6341)

pvizeli committed:
OwnTrack Async (#6363)

* Migrate owntrack to async

* fix tests

pvizeli committed:
Fix possibility that have multible topic subscribe mqtt (#6372)

pvizeli committed:
Migrate mqtt tracker and arwn sensor to async / cleanup owntrack (#6373)

* Migrate mqtt tracker and arwn sensor to async / cleanup owntrack

* Fix tests / lint

balloob committed:
Z-Wave prevent I/O event loop (#6369)

* Prevent Z-Wave I/O in event loop

* Move value_handler to util class.

* Add docstring

valentinalexeev committed:
Update pwaqi to 3.0 to use public API (#6376)

The underlying PWAQI library version 3.0 is now using public API to
access AQICN data.

mezz64 committed:
Update Hikvision Binary Sensors to latest library, remove pyDispatcher (#6231)

* Update pyHik version, remove pyDispatcher in favor of callbacks

* Fix naming

* Fix lint blank line

* Move stream thread start to HOMEASSISTANT_START event

* Bump library version to cleanup shutdown

* Fix requirements

colinodell committed:
Don't initialize components which have already been discovered (#6381)

* Don't initialize components which have already been discovered (fixes #5588)

* Don't log that we've found a service unless we know it's not a duplicate

* Encode discovery data hash with JSON

This also solves the issue of trying to hash non-hashable objects like dicts

* Add test for duplicate device discovery

joe248 committed:
Comed Hourly Pricing sensor (#6378)

* Add ComEd RRTP price sensor

* Update wording to reflect ComEd's naming change from 'RRTP' to 'Hourly Pricing'

* Changed name of sensor source file

* Cleanup based on requested changes

* More cleanup

* small cleanups

Deinara committed:
Resolved issue #5688

Deinara committed:
tweaks

Deinara committed:
Changes for Travis bot. Unused variable 'on_ready' will likely remain reported

Deinara committed:
Edit docstring

titilambert committed:
Add multi contracts support for Hydroquebec (#6392)

andrey-git committed:
Add Zwave refresh services (#6377)

* Add Zwave refresh services

* services file

* Use dispatcher

* Add zwave prefix to signal

aronsky committed:
Add keep-alive feature to the generic thermostat (#6040)

* Add keep-alive feature to the generic thermostat

* Comply with maximum line lengths

* Added tests for the keep-alive functionality

siebert committed:
Fix wake_on_lan for german version of Windows 10 (#6397) (#6398)

Danielhiversen committed:
flux led lib (#6404)

jvolkman committed:
Small typo fix in setup_docker_prereqs

pvizeli committed:
Cleanup async handling (#6388)

* Cleanups unneeded blocks

* Cleanup bootstrap

* dedicated update_ha_state

* Fix imap_email_content

* fx tests

* Fix lint & spell

kellerza committed:
Restore for automation entities (#6254)

* Restore for automation entities

* coroutine

* no clue what i'm doing now

* Still passes nicely in py 3.4

balloob committed:
Fix tests no internet (#6411)

* Fix honeywell tests without internet

* Fix device tracker without internet

* Fix MFI using internet during tests

* Remove I/O from apns tests

balloob committed:
Prevent more I/O in apns (#6413)

pvizeli committed:
Restore flow on device_tracker platform (#6374)

* Restore flow on device_tracker platform

* fix flow

* fix lint

rytilahti committed:
switch.tplink: catch exceptions coming from pyHS100 to avoid flooding the logs when the plug is not available (#6400)

tchellomello committed:
Added sensors to support Ring.com devices (#6419)

titilambert committed:
Add Mint finance sensor (#6132)

* Add Mint finance sensor

* Add retry

* Fix PR comments

* Upgrade mintapi version

* Update mint_finance.py

* Doc tweak

* Update mint_finance.py

balloob committed:
Split bootstrap into bs + setup (#6416)

* Split bootstrap into bs + setup

* Lint

balloob committed:
Tweak recorder/restore_state (#6412)

* Tweak recorder/restore_state

* Lint

balloob committed:
Fix unnecessary warning for ip bans.yaml (#6417)

balloob committed:
Better restore_state warnings (#6418)

amelchio committed:
Set new color before turning LIFX bulbs on (#6402)

A LIFX bulb maintains its previous color even when the light is off.
For example, if the previous color is blue and the bulb is turned on
and then set to a red color, it will transition through purple colors.

After this commit, the target color is set while the bulb is still
turned off. This overrides the previous color and brightness that the
bulb remembered. The light is then turned on with the requested
transition duration.

For the example, this gives the expected result of only going through
red colors.

ishults committed:
Don't log username and password in camera url (#6390)

* Don't log username and password in camera url

* Attempt fix of tox issues

* Attempt to fix indentation issue

amelchio committed:
Ignore deleted mails in IMAP unread count (#6394) (#6395)

Message deletion in IMAP is a two step process: first delete, then expunge.
Deleting a message just sets a flag that usually makes the mail client hide
the message. It is the expunge that actually removes the message.

Thus, exclude the deleted messages so that the unread count matches up with
that of most mail clients.

andrey-git committed:
Delay zwave updates for 100ms to group them. (#6420)

* Add Zwave refresh services

* services file

* Use dispatcher

* Add zwave prefix to signal

* Delay zwave updates for 100ms to group them.

* Fixes

* lint

* Access _scheduled_update from loop thread only.

* More async

* Some optimizations

* Fix

andrey-git committed:
Rename _scheduled_update to _update_scheduled (#6434)

emlove committed:
Revert "Use dynamic port allocation for tests" (#6436)

Commit:
Tado device_tracker exception when mobile device has geofencing enabled but location is currently unknown. (#6401)

andrey-git committed:
Add a Z-wave workaround to do full refresh on update (#6403)

* Add Zwave refresh services

* services file

* Use dispatcher

* Add zwave prefix to signal

* Add a Z-wave workaround to do full refresh on update

dennisdegreef committed:
Use bundled certificates if port matches mqtts (#6429)

* Use bundled certificates if port matches mqtts

* Move import requests.certs to top, since it's used in more places

* Add happy and non-happy path tests for default certificate bundle on mqtts port

pvizeli committed:
Bugfix samsungtv discovery (#6438)

tchellomello committed:
Added unittest for Ring sensor (#6447)

Add an optional extended description…

balloob committed:
Shorten recorder connection init (#6432)

* Wait up to 9 seconds

* Set number of recorder retries to 8

* Do not sleep when reporting last connection error if no retries left

* Make sure we clean up old engine if connection is retrying

* Update __init__.py

bimbar committed:
KWB Easyfire support (#6018)

* KWB Easyfire Support

* requirements, coverage

* Initialization fun

* lint

* requirements bump

* lint

* Second best validation ...

* changes

* reworked validation

Commit:
Bumped version number for supporting lib (#6462)

andersonshatch committed:
Send a logo with webostv notifications (#6380)

* Update to pylgtv 0.1.4

* Send icon with webostv notifications

Default to the homeassistant logo, but allow customizing it on the
component and for individual notifications

balloob committed:
Upgrade netdisco to 0.9.2 (#6466)

balloob committed:
Allow testing against uvloop (#6468)

appzer committed:
fix issue (#6470)

* fix issue

fix issue: https://community.home-assistant.io/t/error-in-new-notification-pushsafer/13308

* Update pushsafer.py

* Update requirements_all.txt

* Update pushsafer.py

* Update pushsafer.py

* Update pushsafer.py

aequitas committed:
Remove connection status state. (#6475)

Current implementation of connection status doesn't follow convention and is not properly configurable. Might be added again in the future as a full fledged entity or some other way.

For now users can rely on error logging to determine connection status.

fronzbot committed:
Support for Blink Camera System (#6444)

* Passing pep8, no tests yet

* Fixed some issues with the request throttling

* Removed ability to set throttle time because it was causing more issues than it was worth

* Added blink to .coveragerc

* Changed blinkpy version

* Removed global var, fixed per PR requests

* Added services for camera, migrated switch to binary_sensor

* Added schema for service, fixed naming, removed unused function

balloob committed:
Add warning for slow platforms/components (#6467)

* Add warning for slow platforms/components

* Add test for slow component setup.

* Add test for slow platform setup

* Fix tests on Py34

siebert committed:
Fix wake_on_lan ping for Linux. (#6480)

pvizeli committed:
Add dispatcher camera for internal image. (#6471)

* Add dispatcher camera for internal image.

* fix lint

* Add unittest

* Update dispatcher.py

pvizeli committed:
Add support for remove services / Reload script support (#6441)

* Add support for remove services / Reload script support

* Reload support for scripts

* Add more unittest for services

* Add unittest for script reload

* Address paulus comments

robbiet480 committed:
Expand MQTT lights (#6481)

* Add effect support to MQTT Light

* Use effect state topic for supported_features

* Dont use rainbow as default color

* Add color_temp support to MQTT JSON Light

* Add effect to MQTT JSON light

* Support lights in MQTT discovery

* Allow discovered devices to set their platform

* Add white value support to MQTT Light

* Add white value support to MQTT JSON Light

* Remove blank line

* Add color_temp support to MQTT Template light

* Add white value support to MQTT Template Light

* Remove unused SUPPORT_MQTT_TEMPLATE and stale unused flash and transition code from MQTT Template

* Add XY Color to MQTT Light Platform

* Fix syntax

* Fix more syntax errors

* Revert "Remove unused SUPPORT_MQTT_TEMPLATE and stale unused flash and transition code from MQTT Template"

This reverts commit c03798cb636e6b3258fce6f676fc6ca34c9b7d37.

* MQTT Template supports flash and transition but doesnt allow templating of the values

* Add XY color support to MQTT JSON

* Proper variable names

* Only allow whitelisted MQTT platforms to be loaded via MQTT Discovery

* Minor tweaks.

colinodell committed:
Allow configurable conditions for Pi-Hole sensor (#6465)

* Allow configurable conditions for Pi-Hole sensor

* Include all three conditions by default

* Share Pi-Hole API data across all sensors; eliminate redundant API calls

reedriley committed:
Improved iCloud 2FA support. (#5984)

pvizeli committed:
Android IP Cam support

arraylabs committed:
Update pymyq requirement (#6486)

* update pymyq to 0.0.7

* update pymyq requirement

Adds additional state for stopped door.

pvizeli committed:
add available

pvizeli committed:
fix lint

balloob committed:
Not always asume manufacturername is present (#6484)

emlove committed:
Catch exceptions on kodi commands

pvizeli committed:
Fix some things

pvizeli committed:
fix coverage

robbiet480 committed:
MQTT should load the provided platform not DOMAIN

See #6481

robbiet480 committed:
Add required changes to support MQTT discovery that I previously missed. See #6481.

pvizeli committed:
pump version 0.3 / make a lot of improvments

pvizeli committed:
fix lint

balloob committed:
Add first pass at Z-Wave light tests (#6483)

* Add first pass at Z-Wave light tests

* Remove unused SIGNAL_VALUE

* Lint

* Update test_init.py

pvizeli committed:
Fix lint

pvizeli committed:
Bugfix mqtt socket memory error (#6501)

* Bugfix mqtt socket memory error

* Fix tests

* Fix lint

amelchio committed:
Increase upper limit on light transitions (#2843)

This increases the global limit to 6535 seconds (1h48m55s) because that is
supported by all light platforms.

Some platforms support even longer transition times so the limit should
actually be platform specific.

happyleavesaoc committed:
more moon states

Deinara committed:
Added pylint ignore message and re-added .close()

Deinara committed:
Added an extra space

emlove committed:
Include functools.wraps

robbiet480 committed:
Merge pull request #6488 from pvizeli/pydroid-ipcam

Android IP Cam support

robbiet480 committed:
Merge pull request #6506 from happyleavesaoc/enhance-moon

more moon states

robbiet480 committed:
Merge pull request #6507 from amelchio/longer-light-transitions

Increase upper limit on light transitions (#2843)

robbiet480 committed:
Merge pull request #6494 from armills/kodi-command-exceptions

Catch exceptions on kodi commands

robbiet480 committed:
Make states constants

robbiet480 committed:
Merge pull request #6407 from jvolkman/patch-1

Small typo fix in setup_docker_prereqs

robbiet480 committed:
Merge pull request #6389 from Deinara/discord_connect_fix

Discord connect fix

robbiet480 committed:
Merge pull request #6361 from GreenTurtwig/dev

Updated to catch timeout error

robbiet480 committed:
Merge pull request #6001 from jumpkick/patch-2

Improvements for WeMo Insight switches

Commit:
Merge remote-tracking branch 'origin/master' into dev

pvizeli committed:
Bugfix android camera autodiscovery settings (#6510)

Add an optional extended description…

craigjmidwinter committed:
Insteon lib (#6505)

* use lib with caching to reduce collisions

* use 0.43

* change requirements

* update the  lib to 0.44

* update req

* fix typo

* just keep checking

* just keep checking - switch

* use 0.45 with file cache

* requirements

* Update requirements_all.txt

* Update insteon_local.py

* Update requirements_all.txt

* Update insteon_local.py

* update library

* fix lint

finish06 committed:
Update to Pyunifi2.0 (#6490)

* Updated pyUnifi

* Missing comma

* Security opt-out, not opt-in

* Adjust minimal values

* Update to pyUnifi 2.0

pvizeli committed:
Bugfix rpi_rf cleanup (#6513)

Add an optional extended description…

tchellomello commented:
@robbiet480  @balloob  any chances  to get the PR https://github.com/home-assistant/home-assistant/pull/6520 (if approved) on 0.40?  Thanks guys!

pvizeli committed:
Android webcam better error handling / pump library 0.4 (#6518)

MartinHjelmare committed:
Fix mysensors gateway windows setup (#6500)

balloob commented:
@tchellomello no more new features for 0.40.

balloob committed:
Update frontend

balloob committed:
Version bump to 0.40

balloob committed:
Remove mint finance sensor (#6522)

Landrash commented:
the package for the rpi-update command isn't included in the base image of raspbian and therefore not in the HASSbian image either.

I've changed this in the next build but currently it needs to be installed with 

```
sudo apt-get install rpi-update
```

arsaboo commented:
@balloob In the PR, you have the command as `rpi-upgrade`. It should be `rpi-update`

pavoni committed:
Append vera device id to entity id - but not name. (#6523)

* Append vera device id to entity id - but not name.

* Tidy.

* Tidy.

* Tidy after review.

* Re-order.

balloob commented:
@Landrash I have linked to the install + update instructions. Hope that will help.

pavoni committed:
Discovery is a dict rather than an array. (#6525)

amelchio committed:
Fix colortemp conversion for osramlightify (#6516)

* Fix colortemp conversion for osramlightify

Copied from the LIFX fix in 75df4be73391d5255371632f5fb08d7251e85b13.

* Fix style

* Updates from review

@armills:

While we're doing cleanup here, can you just change self._brightness,
self._rgb, self._name, self._temperature, and self._state assignments in
__init__ to None? These will get overwritten when self.update() is called, so
it's safer/cleaner to initialize them to None since it shouldn't matter if
everything is working.
